documentation/modules/adding-source-provider.adoc

```diff
@@ -21,35 +21,161 @@ You can add {a-rhv} source provider by using the {ocp} web console.
 
 * {manager} CA certificate, unless it was replaced by a third-party certificate, in which case, specify the {manager} Apache CA certificate
 endif::[]
+ifdef::ova[]
+= Adding an Open Virtual Appliance (OVA) source provider
+
+You can add Open Virtual Appliance (OVA) files that were created by or are compatible with VMware vSphere as source providers.
+
+[IMPORTANT]
+====
+Migration using Open Virtual Application (OVA) files that were created by or are compatible with VMware vSphere as source providers is a Technology Preview feature only. Technology Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete. Red Hat does not recommend using them in production. These features provide early access to upcoming product
+features, enabling customers to test functionality and provide feedback during the development process.
+For more information about the support scope of Red Hat Technology Preview
+features, see https://access.redhat.com/support/offerings/techpreview/.
+====
+endif::[]
+ifdef::ostack[]
+= Adding an {osp} source provider
+
+You can add an {osp} source provider by using the {ocp} web console.
+
+[NOTE]
+====
+Migration using {osp} source providers only supports VMs that use only Cinder volumes.
+====
+endif::[]
+ifdef::cnv[]
+= Adding a Red Hat {virt} source provider
+
+You can add a Red Hat {virt} source provider by using the {ocp} web console.
+
+.Prerequisites
+
+* TBD
+endif::[]
+ifdef::cnv2[]
+= Adding {a-virt} destination provider
+
+You can add {a-virt} destination provider to the {ocp} web console in addition to the default {virt} destination provider, which is the provider where you installed {project-short}.
+
+.Prerequisites
+
+* You must have {a-virt} link:https://access.redhat.com/documentation/en-us/openshift_container_platform/{ocp-version}/html/authentication_and_authorization/using-service-accounts[service account token] with `cluster-admin` privileges.
+endif::[]
 
 .Procedure
 
 . In the {ocp} web console, click *Migration* -> *Providers for virtualization*.
 . Click *Create Provider*.
 ifdef::vmware[]
-. Select *VMware* from the *Provider type* list.
+. Click *vSphere*.
 . Specify the following fields:
 
-* *Provider name*: Name to display in the list of providers
-* *vCenter host name or IP address*: vCenter host name or IP address - if a certificate for FQDN is specified, the value of this field needs to match the FQDN in the certificate
-* *vCenter user name*: vCenter user, for example, `user@vsphere.local`
-* *vCenter password*: vCenter user password
-* *VDDK init image*: VDDKInitImage path
+* *Provider Resource Name*: Name to display in the list of providers.
+* *URL*: Endpoint.
+* *VDDK Init Image*: `VDDKInitImage` path. It is strongly recommended to create a VDDK init image to accelerate migrations.
```

Comment:
the wording looks good to me, can we also link to the `Creating a VDDK image` section here?

Reply:
yes, we have a bug for adding more information in hints:
I'll add this to:
https://github.com/kubev2v/forklift-console-plugin/issues/541

Reply:
@yaacov that would be great but I meant adding the link to the documentation :)

Reply:
This should work:
```
xref:../master.adoc#creating-vddk-image_mtv[Creating a VDDK image]
```
Please double-check @RichardHoch

Reply:
@anarnold97 Yes!